### PR TITLE
Move 0.8.23 into 0.9.0 feature branch

### PR DESCRIPTION
**Pull request by awalkowiak**
I cherry-picked all the commits from the 0.8.23 to the feature 0.9.0 branch. 

I also did some required fixes for the new version:
- Moved new Config.groovy configurations to the runtime.groovy
- Moved some files to proper directory
- Moved RequisitionStatusTransitionEventService to proper package
- Replaced transactional = true with [at]Transactional
- Replaced old ConfigurationHolder and ApplicationHolder with grails.util.Holders

**DO NOT SQUASH THIS PR**

### CODE REVIEW

**Review by alannadolny — APPROVED** [review]
Looks like everything is included, lgmt

File: grails-app/conf/Config.groovy
Comment (drodzewicz) [inline grails-app/conf/Config.groovy:1]: Whats up withe empty `Config.groovy` file?

Comment (awalkowiak) [reply grails-app/conf/Config.groovy:1]: Should be removed, did not notice it

**Review by kchelstowski — APPROVED** [review]
lgtm except the empty Config.groovy

**Pull request merged**